consistent_hash/hash.go

```diff
@@ -0,0 +1,68 @@
+package consistent_hash
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"sort"
+	"strconv"
+)
+
+type Node struct {
+	key  string
+	hash uint64
+}
+
+type HashRing struct {
+	nodes    []*Node
+	Replicas int
+}
+
+func NewHashRing(replicas int) *HashRing {
+	return &HashRing{
+		Replicas: replicas,
+	}
+}
+
+func (hr *HashRing) AddNode(key string) {
```

Comment:
### Runtime Errors

- **Check for nil pointers**: The `AddNode` and `RemoveNode` methods should ensure that `hr` is not nil to avoid panics.

```go
func (hr *HashRing) AddNode(key string) {
    if hr == nil {
        panic("HashRing is nil")
    }
    // existing code
}
```

```go
func (hr *HashRing) RemoveNode(key string) {
    if hr == nil {
        panic("HashRing is nil")
    }
    // existing code
}
```